patches/tModLoader/Terraria/Program.cs.patch

```diff
@@ -109,12 +109,16 @@
  			}
  		}
  
-@@ -202,9 +_,18 @@
+@@ -202,13 +_,43 @@
  				}
  
  #if CLIENT
 -				MessageBox.Show(e.ToString(), "Terraria: Error");
-+				ModLoader.UI.Interface.MessageBoxShow(e.ToString());
```

Comment:
I suggest the following:
`ModLoader.UI.Interface.MessageBoxShow(e);` 
and have it enhance the exception automatically.
Making a method outside the class in one of our own reduces the patch, and possible margin of error
A method that enhances the error is the case here, with a simple change of calling our own MessageBox instead of vanillas'

Reply:
Done - I created a `MessageBoxShow` overload that takes an `Exception` and an optional `generateTip` parameter. By default, it calls the original `MessageBoxShow` with the exception message, but if `generateTip` is true then it'll also show a tip. Hopefully this is what you meant 👍

Reply:
It's probably better if MessageBoxShow accepts a string parameter, then make a 'GenerateTip()' method that returns the standard tips. This allows custom strings when needed.